templates/postgresql_database.yml.erb

```diff
@@ -4,9 +4,14 @@ development: &default
   encoding: utf8
   host: localhost
   min_messages: warning
-  pool: 2
+  pool: <%%= ENV.fetch("DB_POOL", 2) %>
+  reaping_frequency: <%%= ENV.fetch("DB_REAPING_FREQUENCY", 10) %>
   timeout: 5000
 
 test:
   <<: *default
   database: <%= app_name %>_test
+
+production:
+  url:  <%%= ENV.fetch("DATABASE_URL") %>
+  pool: <%%= ENV.fetch("DB_POOL", ENV.fetch("MAX_THREADS", 5) %>
```

Comment:
You need your pool to at least be as large as your puma thread count. Btw I recommend against using a "staging" specific environment: https://devcenter.heroku.com/articles/deploying-to-a-custom-rails-environment